src/Network/Network/Bastion/BastionBaseCmdlet.cs

```diff
@@ -0,0 +1,103 @@
+﻿// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Net;
+using AutoMapper;
+using Microsoft.Azure.Commands.Network.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.Tags;
+using Microsoft.Azure.Management.Network;
+using MNM = Microsoft.Azure.Management.Network.Models;
+
+namespace Microsoft.Azure.Commands.Network.Bastion
+{
+    public abstract class BastionBaseCmdlet : NetworkBaseCmdlet
+    {
+        public IBastionHostsOperations BastionClient
+        {
+            get
+            {
+                return NetworkClient.NetworkManagementClient.BastionHosts;
+            }
+        }
+        
+        protected IVirtualNetworksOperations VirtualNetworkClient
+        {
+            get
+            {
+                return NetworkClient.NetworkManagementClient.VirtualNetworks;
+            }
+        }
+
+        protected IPublicIPAddressesOperations PublicIPAddressesClient
+        {
+            get
+            {
+                return NetworkClient.NetworkManagementClient.PublicIPAddresses;
+            }
+        }
+
+        public bool IsBastionPresent(string resourceGroupName, string name)
```

Comment:
Please use `IsResourcePresent` from `NetworkBaseCmdlet`. Example of usage: https://github.com/Azure/azure-powershell/blob/network-september/src/Network/Network/Cortex/VirtualWan/VirtualWanBaseCmdlet.cs#L110

Reply:
Updated this to IsResourcePresent.